src/hooks/useConnectionForm.ts

```diff
@@ -197,6 +197,15 @@ export const useConnectionForm = (validationSchema: ZodObject<ZodRawShape>, mode
 			);
 			navigate(`/projects/${projectId}/connections`);
 		} catch (error) {
+			if (axios.isAxiosError(error)) {
+				LoggerService.error(
+					namespaces.hooks.connectionForm,
+					tErrors("errorEditingNewConnectionExtended", { error: error?.response?.data })
+				);
+				setIsLoading(false);
```

Comment:
move `setIsLoading(false); `to` finally{setIsLoading(false)}`

Reply:
Refactored and added more changes: https://github.com/autokitteh/web-platform/pull/622/commits/7ecaaa3c6ce9ca720e518ccfc0ddf7079c6aa110